### PR TITLE
Adding type='password' support for Text in TextView class.

### PR DESCRIPTION
**Pull request by jakhani**
Currently there is no way to use Text widget for Password. This code will allow to pass type while using Text widget which will allow to set type='password' for Text.

### CODE REVIEW

**Comment by jasongrout** [comment]
Thanks! Up to now, we've made different types of html widgets different things (e.g., the checkbox, color, radio, etc.). Following that pattern, we'd have a `Password()` widget. What do you think about making a separate `Password()` widget that inherits from the text widget? On the other hand, I can see this just being basically just tweaking a textbox, so perhaps making a password=True/False attribute on a textbox would work well.

**Comment by jakhani** [comment]
@jasongrout This is to follow the way html works. In HTML, when you pass type="password", it doesn't show it's value. If there is any other way already available, can you please share some sample code with me?

**Comment by jasongrout** [comment]
@jakhani - I agree it would be great to make this available to users, and there's no way to do it right now. What I'm suggesting is that we either (a) make a new `Password()` that acts almost exactly like a text widget, except that in the view we set the type to password, or (b) make a single `password` attribute to the `Text()` widget that controls this switch. Each of those options makes it easy to expose this one thing (a password box) in a pythonic api.

**Comment by jasongrout** [comment]
(while exposing the input `type` exposes all of the sorts of inputs HTML has to offer: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input)

**Comment by jakhani** [comment]
@jasongrout Sure. I have added new widget Password. I have tested it and it's working fine. Can you please review this code?

File: ipywidgets/widgets/widget_string.py [hunk 1]
```diff
@@ -107,3 +107,40 @@ def on_submit(self, callback, remove=False):
             Whether to unregister the callback
         """
         self._submission_callbacks.register_callback(callback, remove=remove)
+
+
+@register
+class Password(_String):
```
Comment (jasongrout) [inline ipywidgets/widgets/widget_string.py]: Would it be easier to just inherit from Text?

```
@register
class Password(Text):
    _view_name = ...
    _model_name = ...
```
Then you inherit all of the text behavior automatically.

File: jupyter-js-widgets/src/widget_string.ts [hunk 2]
```diff
@@ -385,3 +385,121 @@ class TextView extends LabeledDOMWidgetView {
...
+    defaults() {
+        return _.extend(super.defaults(), {
+            _view_name: 'PasswordView',
+            _model_name: 'PasswordModel'
+        });
+    }
+}
+
+export
+class PasswordView extends LabeledDOMWidgetView {
```
Comment (jasongrout) [inline jupyter-js-widgets/src/widget_string.ts]: Again, here how about just inheriting from `TextView`, and doing something like:

```
render() {
    super.render();
    this.textbox.setAttribute('type', 'password');
}
```
and let everything else be inherited?

Comment (jasongrout) [inline jupyter-js-widgets/src/widget_string.ts]: (the only issue I could see here is if the element is appended to the DOM before the input is set to password, the default string might flash up on the screen exposed)

Comment (jasongrout) [inline jupyter-js-widgets/src/widget_string.ts]: (i.e., that's something we should check to see if it happens)

Comment (jasongrout) [inline jupyter-js-widgets/src/widget_string.ts]: In fact, why don't we change `TextView` to have a new protected attribute (at the bottom of the class), the `inputType`, which for `TextView` is `'text'` and is used in the `render()` method, but the `PasswordView` subclass just sets this to `'password'`. Then the PasswordView just becomes something like:

```typescript
export
class PasswordView extends TextView {
    inputType: string = 'password';
}
```
or something like that.

**Comment by jakhani** [comment]
@jasongrout I have incorporated all your comments and submitted changes. Can you please have a look at it?

File: jupyter-widgets-schema/jupyterwidgetmodels.md [hunk 3]
```diff
@@ -661,6 +661,23 @@ Attribute        | Type             | Default          | Help
 `placeholder`    | string           | `'\u200b'`       | Placeholder text to display when nothing has been typed
 `value`          | string           | `''`             | String value
 
+### Jupyter.Password
```
Comment (jasongrout) [inline jupyter-widgets-schema/jupyterwidgetmodels.md]: Can you delete the changes to this file? This file is documenting the widgets as of 2.1.0 and is automatically generated. We'll generate another listing when we release the next version.

**Comment by jasongrout** [comment]
> @jasongrout I have incorporated all your comments and submitted changes. Can you please have a look at it?

I made a few small changes (reformatting, and changing the inputType to also be readonly), and left one more comment. I'll test this tomorrow, but the code looks good to me now. Feel free to squash in my changes if you want, or leave them as separate commits - either way is fine.

Thanks again!

**Comment by jakhani** [comment]
I have removed schema file changes. One check is failing and I am not able to get logs for the same. Can you please share detail on that in case you find anything?

**Comment by jasongrout** [comment]
It seems like there is some problem with the logs on Travis right now. Hopefully my fix fixes whatever the error was.

**Comment by jasongrout** [comment]
Thank you very much!

**Comment by jakhani** [comment]
@jasongrout Thanks for helping and merging it quickly.

**Comment by jasongrout** [comment]
And congratulations on your first contribution to ipywidgets!

**Comment by maartenbreddels** [comment]
Indeed, thanks for this contribution! 👍

**Comment by jasongrout** [comment]
I thought this was in the alpha version - can you use our alpha release?

We are working hard on finishing up what is needed for the 7.0 final release. If you need this widget before 7.0 is released, you can make a new python package with just this widget and use that in the meantime, right?

**Pull request merged**